File: 2-medium/00287-find-the-duplicate-number/solution.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {

        int num{0};
        for (int i{0}; i <= nums.size(); ++i) {
            num = abs(nums[i]);
            if (nums[num] > 0) {
                nums[num] *= -1;
            } else {
                break;
            }
        }

        for (int i{0}; i < nums.size(); ++i) {
            if (nums[i] < 0) {
                nums[i] *= -1;
            }
        }
        return num;
    }
};

File: 2-medium/00287-find-the-duplicate-number/solution.rs
impl Solution {
    pub fn find_duplicate(mut nums: Vec<i32>) -> i32 {

        let mut num = 0;

        for i in 0..nums.len() {
            num = nums[i].abs();
            if nums[num as usize] > 0 {
                nums[num as usize] *= -1;
            } else {
                break;
            }
        }

        for i in 0..nums.len() {
            if nums[i] < 0 {
                nums[i] *= -1;
            }
        }

        num
    }
}
